xcode/Subconscious/Shared/Models/Did.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  Did.swift
+//  Subconscious
+//
+//  Created by Ben Follington on 1/3/2023.
+//
+
+import Foundation
+
+struct Did : Hashable, Identifiable {
+    var did: String
+    var id: String { did }
+    
+    // Approximate, based on https://www.w3.org/TR/did-core/#did-syntax
+    static let regex = /^(did:[a-z0-9]{3}:[a-zA-Z0-9-_\.%:]+)$/
+}
+
+extension Did {
```

Comment:
Because we can parse losslessly from a string, this extension should conform Did to `LosslessStringConvertible` https://developer.apple.com/documentation/swift/losslessstringconvertible.

The only thing you would have to add is a description property.

```swift
extension Did: LosslessStringConvertible {
    var description: { did }
    // ...
}
```